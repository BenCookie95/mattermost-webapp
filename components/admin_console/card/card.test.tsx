// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage} from 'react-intl';
import {shallow} from 'enzyme';

import {TestHelper} from 'utils/test_helper';

import Card, {Props} from './card';

describe('components/admin_console/card', () => {
    const defaultProps: Props = {
        title: 
            <FormattedMessage
                id='admin.data_retention.customPolicies.title'
                defaultMessage='Custom retention policies'
            />
        ,
        subtitle: 
            <FormattedMessage
                id='admin.data_retention.customPolicies.subTitle'
                defaultMessage='Customize how long specific teams and channels will keep messages.'
            />
        ,
        body: 
            <>
                Hello!
            </>
        ,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<Card {...defaultProps}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show button in card header', () => {
        const wrapper = shallow(<Card {...defaultProps}/>);

        wrapper.setProps({
            buttonText: 
                <FormattedMessage
                    id='admin.data_retention.customPolicies.addPolicy'
                    defaultMessage='Add policy'
                />
            ,
            onClick:
                () => {
                    console.log('Test');
                }
            ,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
