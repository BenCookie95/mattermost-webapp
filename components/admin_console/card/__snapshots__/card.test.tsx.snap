// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/card should match snapshot 1`] = `
<div
  className="Card__card"
>
  <div
    className="Card__cardHeader"
  >
    <div
      className="Card__cardHeaderText"
    >
      <div
        className="Card__cardHeaderText-top"
      >
        <FormattedMessage
          defaultMessage="Custom retention policies"
          id="admin.data_retention.customPolicies.title"
        />
      </div>
      <div
        className="Card__cardHeaderText-bottom"
      >
        <FormattedMessage
          defaultMessage="Customize how long specific teams and channels will keep messages."
          id="admin.data_retention.customPolicies.subTitle"
        />
      </div>
    </div>
  </div>
  <div
    className="Card__cardBody"
  >
    Hello!
  </div>
</div>
`;

exports[`components/admin_console/card should show button in card header 1`] = `
<div
  className="Card__card"
>
  <div
    className="Card__cardHeader"
  >
    <div
      className="Card__cardHeaderText"
    >
      <div
        className="Card__cardHeaderText-top"
      >
        <FormattedMessage
          defaultMessage="Custom retention policies"
          id="admin.data_retention.customPolicies.title"
        />
      </div>
      <div
        className="Card__cardHeaderText-bottom"
      >
        <FormattedMessage
          defaultMessage="Customize how long specific teams and channels will keep messages."
          id="admin.data_retention.customPolicies.subTitle"
        />
      </div>
    </div>
    <button
      className="Card__cardHeaderButton primary"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Add policy"
        id="admin.data_retention.customPolicies.addPolicy"
      />
    </button>
  </div>
  <div
    className="Card__cardBody"
  >
    Hello!
  </div>
</div>
`;
